### ISSUE TITLE
"search in description" doesn't work

### ISSUE THREAD

**Issue by tposney** [tposney/midi-qol#243]
In GitLab by @stefanu91 on Feb 16, 2021, 06:45

Does not read the description in the spell, and always applies "no damage to the TS"

**Comment by tposney** [comment]
That's not correct. I can't begin to guess what you are doing but an example would really help.

**Comment by tposney** [comment]
In GitLab by @stefanu91 on Feb 17, 2021, 21:49

Excuse me Tim if I couldn't explain correctly.
I'm not very familiar with English ...
I am attaching a gif for you to see.
Basically when a new spell is created, and it is written in the description of it "no damage to the save" midi does not read the description and pulls them always doing "no damage to the save"
Sorry again for the inconvenience and thanks for the reply.![Test_save](https://my-gitlab-bucket.s3.amazonaws.com/640711674/79c0a46da20e0674101278217d306dab90a168e74ffc07d041cdf24d0b73d271/Test_save.gif)

**Comment by tposney** [comment]
Thanks that is very helpful and your English is fine.
For a bit of background. When you choose search spell description, every item that is of type saving throw becomes 0 damage on save (instead of 1/2 damage on save) UNLESS the specific text "half as much damage" or "half damage" is found in the text description. 

I just had a look and that test is case sensitive (which feels like a bug to me - so I will change it to not be case insensitive).

If you try "half damage" it should work as you expect and I will remove the case sensitive requirement in the next version.

And sorry if I was snappy in my initial response. I get lots of "X does not work" issues and it can be very difficult to work out what is going wrong when people don't give me any context about the issue. I generally test all of the core features of the module to make sure they work, but can't cover every case people have and can't really afford to spend lots of time trying to deduce the way they are trying to use the module. So reports that don't give me anything to work with are frustrating since I can't really do much with them. Enough detail for me to work out what people are trying to do makes life much easier.

**Comment by tposney** [comment]
In GitLab by @stefanu91 on Feb 17, 2021, 23:54

I will never end to thank you !!!

I found that the problem only occurs with the Italian language.
Before it also worked in my language, but it's not a serious problem anyway, as long as I put the English language back and everything goes back to normal.

I always try to solve a problem with my own strength, but I don't always succeed xD

**Comment by tposney** [comment]
Hmm, it is supposed to check the localised strings (if you are using a
non-english setup).
    "midi-qol.halfDamage": "la metà di quei danni",
    "midi-qol.halfDamageAlt": "la metà dei danni",

I know that this works for the japanese users, but have not checked
Italian. The case sensitive issue will still exist (until the next update),
but it should be checking you local language.
Can you try either of those strings (excactly) and see if it works?

**Comment by tposney** [comment]
In GitLab by @stefanu91 on Feb 18, 2021, 24:32

I don't think I understood correctly, but I tried all the possible words in Italian in the description of the object.
(attached gif of the test)![Test_save_ita_2](https://my-gitlab-bucket.s3.amazonaws.com/640711674/c11875bd3af6e0dd380d9ff740f920e5d2d05070fddde25bdc94b92329a29c63/Test_save_ita_2.gif)

**Comment by tposney** [comment]
In GitLab by @stefanu91 on Feb 18, 2021, 24:35

I am noticing that the problem only arises with the spells or weapons I create, or when I change the description of an item.
(even if I just change a comma)

**Comment by tposney** [comment]
Can you send me the item? I want to capture exactly the text you have to
see why the comparison is not working. It should support what you are
trying to do. (Export the item to json from the sidebar).

**Comment by tposney** [comment]
In GitLab by @stefanu91 on Feb 18, 2021, 01:56

[fvtt-Item-test_spell.json](https://my-gitlab-bucket.s3.amazonaws.com/640711674/55be3c1e196fa7634e8187ee7ff333584897786dfbbff2baa57528083ed925c9/fvtt-Item-test_spell.json)

**Comment by tposney** [comment]
In GitLab by @stefanu91 on Feb 18, 2021, 02:07

undefined

**Comment by tposney** [comment]
In GitLab by @stefanu91 on Feb 18, 2021, 02:28

[fvtt-Item-test_spell1.json](https://my-gitlab-bucket.s3.amazonaws.com/640711674/47224caafc83498d71bbd422b4673b7b3e3028ae9886f0e391b2105a5203ab67/fvtt-Item-test_spell1.json)

this is the object with description

**Comment by tposney** [comment]
Thanks for that. I know what is going on now. The foundry representation of the string "la metà di quei danni" in the description is ```"la met&agrave; di quei danni"``` and the two do not match when checking. It's odd that this has not come up before, since there are quite a few translations with accented characters.

I've tried manually updating the description field (without going through the edtior) and it works as expected. So for some reason the text editor is doing the translation of the UTF character. That also explains why spells stop working if you edit the description, the saved version does not match anymore because of the encoding by the editor.

I've added some code for the conversion back to utf-8 strings on the description and that seems to work. Will go out in the next release of midi-qol.
